src/client/Microsoft.Identity.Client/Internal/Requests/CredentialBasedMsiAuthRequest.cs

```diff
@@ -0,0 +1,300 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Cache.Items;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Credential;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.PlatformsCommon.Interfaces;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client.Internal.Requests
+{
+    internal class CredentialBasedMsiAuthRequest : RequestBase
```

Comment:
I think we need to avoid having another request. There is just too much dup code. 

We can either reuse the ClientCredentials request, which we decided is not easily possible because we can't use CCA here. Or go back to your original design which used "LegacyMsiAuthRequest".

Reply:
CC @pmaytak for an opinion on this.

Reply:
Create a base managed identity request class with common code, legacy and this class can extend it? Can also aid in readability to show what's specific to each MSI request implementation.